File: dayEight/python/TaskTwo/string_of_brackets.py
def allBracketAreClosed(characters) :
    if(isCurlyBracket(characters) or isDiamond(characters) or isParenthesis(characters) or isSquareBracket(characters)) :
        return True;

    return False;

def isCurlyBracket(characters) :
    count = 0;
    for index in range(len(characters)):
        if(characters[index] == '{'):
            count += 1
        
        if(characters[index] == '}'):
            count += 1
            
    return count == 2;

def isSquareBracket(characters):
    count = 0;
    for index in range(len(characters)) :
        if(characters[index] == '['):
            count += 1
        
        if(characters[index] == ']'):
            count += 1

    return count == 2;

def isParenthesis(characters) :
    count = 0;
    for index in range(len(characters)) :
        if(characters[index] == '(') :
            count += 1
        
        if(characters[index] == ')') :
            count += 1
        
    
    
    return count == 2;

def isDiamond(characters) :
    count = 0;
    for index in range(len(characters)) :
        if(characters[index] == '<') :
            count += 1
        
        if(characters[index] == '>') :
            count += 1
        
    
    
    return count == 2;

File: dayEight/javaScript/StringOfBrackets.js

    function allBracketAreClosed(characters){
        if(isCurlyBracket(characters) || isDiamond(characters) || isParenthesis(characters) || isSquareBracket(characters)){
            return true;
        }
        return false;
    }
    function isCurlyBracket( characters){
        let count = 0;
        for(let index = 0; index < characters.length(); index++){
            if(characters.charAt(index) == '{'){
                count++;
            }
            if(characters.charAt(index) == '}'){
                count++;
            }
        }
        
        return count == 2;
    }
    function isSquareBracket( characters){
        let count = 0;
        for(let index = 0; index < characters.length(); index++){
            if(characters.charAt(index) == '['){
                count++;
            }
            if(characters.charAt(index) == ']'){
                count++;
            }
        }
        
        return count == 2;
    }
    function isParenthesis( characters){
        let count = 0;
        for(let index = 0; index < characters.length(); index++){
            if(characters.charAt(index) == '('){
                count++;
            }
            if(characters.charAt(index) == ')'){
                count++;
            }
        }
        
        return count == 2;
    }
    function isDiamond( characters){
        let count = 0;
        for(let index = 0; index < characters.length(); index++){
            if(characters.charAt(index) == '<'){
                count++;
            }
            if(characters.charAt(index) == '>'){
                count++;
            }
        }
        
        return count == 2;
    }
